quickjs.h

```diff
@@ -845,8 +846,40 @@ typedef enum JSTypedArrayEnum {
     JS_TYPED_ARRAY_FLOAT64,
 } JSTypedArrayEnum;
 
+static inline int JS_BytesPerElement(JSTypedArrayEnum type)
+{
+    switch (type) {
+    case JS_TYPED_ARRAY_UINT8C:
+    case JS_TYPED_ARRAY_INT8:
+    case JS_TYPED_ARRAY_UINT8:
+        return 1;
+    case JS_TYPED_ARRAY_INT16:
+    case JS_TYPED_ARRAY_UINT16:
+        return 2;
+    case JS_TYPED_ARRAY_INT32:
+    case JS_TYPED_ARRAY_UINT32:
+    case JS_TYPED_ARRAY_FLOAT32:
+        return 4;
+    default:
+        return 8;
+    }
+}
+
 JSValue JS_NewTypedArray(JSContext *ctx, int argc, JSValueConst *argv,
                          JSTypedArrayEnum array_type);
+
```

Comment:
For consistency with `JS_GetTypedArray`, you might want to change `JS_NewTypedArray` to take a `size_t length` and a `const void *data` in addition to the `JSTypedArrayEnum array_type` and drop the `argc` / `argv` arguments.

Reply:
See https://github.com/bellard/quickjs/pull/294